docs_src/design/adr/0026-Common Configuration.md

```diff
@@ -0,0 +1,393 @@
+  [MessageQueue.Optional]
+  # Default MQTT & NATS Specific options that need to be here to enable environment variable overrides of them
+  ClientId = "device-mqtt"
+  [MessageQueue.Topics]
+  CommandRequestTopic = "edgex/device/command/request/device-mqtt/#"   # subscribing for inbound command requests
+  CommandResponseTopicPrefix = "edgex/device/command/response"   # publishing outbound command responses; <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
+
+[MQTTBrokerInfo]
+Schema = "tcp"
+Host = "localhost"
+Port = 1883
+Qos = 0
+KeepAlive = 3600
+ClientId = "device-mqtt"
+
+CredentialsRetryTime = 120 # Seconds
+CredentialsRetryWait = 1 # Seconds
+ConnEstablishingRetry = 10
+ConnRetryWaitTime = 5
+
+# AuthMode is the MQTT broker authentication mechanism. Currently, "none" and "usernamepassword" is the only AuthMode supported by this service, and the secret keys are "username" and "password".
+AuthMode = "none"
+CredentialsPath = "credentials"
+
+# Comment out/remove when using multi-level topics
+IncomingTopic = "DataTopic"
+ResponseTopic = "ResponseTopic"
+UseTopicLevels = false
+
+# Uncomment to use multi-level topics
+# IncomingTopic = "incoming/data/#"
+# ResponseTopic = "command/response/#"
+# UseTopicLevels = true
+
+    [MQTTBrokerInfo.Writable]
+    # ResponseFetchInterval specifies the retry interval(milliseconds) to fetch the command response from the MQTT broker
+    ResponseFetchInterval = 500
+```
+
+### Modules and Services Impacted
+
+The following modules and services are impacted:
+
+- go-mod-bootstrap
+    - All service configuration bootstrapping is handled in this module
+- All services which consume go-mod-bootstrap
+    - The configuration TOML files will be updated to remove all common  settings
+    - This leaves only the private settings and overrides of certain common settings (See examples above)
+- New service which provides the common configuration
+    - Contains new configuration TOML which contains all the common settings (See example above) 
+    - No environment overrides will be applied to the common settings
+    - This service is responsible for 
+      - Pushing the common settings into the Configuration Provider during first start-up
+      - Providing a REST API to pull the common settings for services not using the Configuration Provider
+- TAF test scripts may be impacted based on which settings are modified via the Configuration Provider
+
+## Considerations
+
+- New service which owns the common configuration could be integrated into Core Metadata rather than creating a new micro service. This violates the micro service design principle of single responsibility.
+- Overrides pushed into Configuration Provider rather than applying the overrides once configuration is pulled from the Configuration Provider . Consensus is that the overridden values should not be stored in the Configuration Provider. 

```

Comment:
```suggestion
- Env. Overrides pushed into Configuration Provider rather than applying the overrides once configuration is pulled from the Configuration Provider . Consensus is that the overridden values should not be stored in the Configuration Provider. 
```
So not to be confused with overrides set in configuration files as described in before:
> The change will be ignored if the setting exists in the service's private section. This is so that the service's private overrides are always retained.

Reply:
done